src/pevm.rs

```diff
@@ -60,15 +62,23 @@ pub fn execute<S: Storage + Send + Sync>(
     let Some(block_env) = get_block_env(&block.header) else {
         return Err(PevmError::MissingHeaderData);
     };
-    let Some(tx_envs) = get_tx_envs(&block.transactions) else {
+    let tx_envs: Vec<_> = block.transactions.txns().map(get_tx_env).collect();
+    if tx_envs.is_empty() {
```

Comment:
This is a major regression:
- Empty blocks are valid.
- `MissingTransactionData` is about having at least an "invalid" transaction, not having no transactions. Perhaps we can give it a better name.
- One of our current goals is to replace `unwrap`s with grateful errors returned to users. This negates one and introduces more panics instead...